### PR TITLE
Add support for v1 k8s volume snapshot APIs

### PR DESCRIPTION
**Pull request by PrasadG193**
Signed-off-by: Prasad Ghangal <prasad.ghangal@gmail.com>

## Change Overview

Add support for v1 k8s volume snapshot APIs

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [x] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

#### On cluster with Alpha snapshot APIs

```
$ go test -v -check.vv .
=== RUN   Test
START: snapshot_test.go:82: SnapshotTestSuite.SetUpSuite
snapshotClass GET do-block-storage
snClass:: 
Driver:: 
PASS: snapshot_test.go:82: SnapshotTestSuite.SetUpSuite 6.579s

START: snapshot_test.go:865: SnapshotTestSuite.TestCreateFromSourceAlpha
PASS: snapshot_test.go:865: SnapshotTestSuite.TestCreateFromSourceAlpha 0.003s

START: snapshot_test.go:909: SnapshotTestSuite.TestCreateFromSourceBeta
PASS: snapshot_test.go:909: SnapshotTestSuite.TestCreateFromSourceBeta  0.001s

START: snapshot_test.go:953: SnapshotTestSuite.TestCreateFromSourceStable
PASS: snapshot_test.go:953: SnapshotTestSuite.TestCreateFromSourceStable        0.001s

START: snapshot_test.go:567: SnapshotTestSuite.TestGetVolumeSnapshotClassFake
PASS: snapshot_test.go:567: SnapshotTestSuite.TestGetVolumeSnapshotClassFake    0.005s

START: snapshot_test.go:504: SnapshotTestSuite.TestNewSnapshotter
PASS: snapshot_test.go:504: SnapshotTestSuite.TestNewSnapshotter        0.001s

START: snapshot_test.go:332: SnapshotTestSuite.TestVolumeSnapshotAlpha
VolumeSnapshot test - source namespace: snapshot-test-source-59290 - target namespace: snapshot-test-target-59290
PASS: snapshot_test.go:332: SnapshotTestSuite.TestVolumeSnapshotAlpha   40.127s

START: snapshot_test.go:343: SnapshotTestSuite.TestVolumeSnapshotBeta
SKIP: snapshot_test.go:343: SnapshotTestSuite.TestVolumeSnapshotBeta (No Storageclass with CSI provisioner, install CSI and create a storageclass for it)

START: snapshot_test.go:177: SnapshotTestSuite.TestVolumeSnapshotClassCloneFake
PASS: snapshot_test.go:177: SnapshotTestSuite.TestVolumeSnapshotClassCloneFake  0.004s

START: snapshot_test.go:240: SnapshotTestSuite.TestVolumeSnapshotCloneFake
PASS: snapshot_test.go:240: SnapshotTestSuite.TestVolumeSnapshotCloneFake       0.005s

START: snapshot_test.go:133: SnapshotTestSuite.TestVolumeSnapshotFake
PASS: snapshot_test.go:133: SnapshotTestSuite.TestVolumeSnapshotFake    0.003s

START: snapshot_test.go:354: SnapshotTestSuite.TestVolumeSnapshotStable
SKIP: snapshot_test.go:354: SnapshotTestSuite.TestVolumeSnapshotStable (No Storageclass with CSI provisioner, install CSI and create a storageclass for it)

START: snapshot_test.go:128: SnapshotTestSuite.TearDownSuite
PASS: snapshot_test.go:128: SnapshotTestSuite.TearDownSuite     2.918s

OK: 9 passed, 2 skipped
--- PASS: Test (49.65s)
PASS
ok      github.com/kanisterio/kanister/pkg/kube/snapshot        49.675s

```

#### On cluster with Stable/Beta Snapshot APIs

```
$ go test -v -check.vv .
=== RUN   Test
START: snapshot_test.go:82: SnapshotTestSuite.SetUpSuite
snapshotClass GET csi-rbdplugin-snapclass
snapshotClass GET csi-rbdplugin-snapclass
snClass:: csi-rbdplugin-snapclass
Driver:: kube-rook-ceph.rbd.csi.ceph.com
PASS: snapshot_test.go:82: SnapshotTestSuite.SetUpSuite 11.787s

START: snapshot_test.go:865: SnapshotTestSuite.TestCreateFromSourceAlpha
PASS: snapshot_test.go:865: SnapshotTestSuite.TestCreateFromSourceAlpha 0.001s

START: snapshot_test.go:909: SnapshotTestSuite.TestCreateFromSourceBeta
PASS: snapshot_test.go:909: SnapshotTestSuite.TestCreateFromSourceBeta  0.001s

START: snapshot_test.go:953: SnapshotTestSuite.TestCreateFromSourceStable
PASS: snapshot_test.go:953: SnapshotTestSuite.TestCreateFromSourceStable        0.001s

START: snapshot_test.go:567: SnapshotTestSuite.TestGetVolumeSnapshotClassFake
PASS: snapshot_test.go:567: SnapshotTestSuite.TestGetVolumeSnapshotClassFake    0.005s

START: snapshot_test.go:504: SnapshotTestSuite.TestNewSnapshotter
PASS: snapshot_test.go:504: SnapshotTestSuite.TestNewSnapshotter        0.000s

START: snapshot_test.go:332: SnapshotTestSuite.TestVolumeSnapshotAlpha
SKIP: snapshot_test.go:332: SnapshotTestSuite.TestVolumeSnapshotAlpha (No Storageclass with CSI provisioner, install CSI and create a storageclass for it)

START: snapshot_test.go:343: SnapshotTestSuite.TestVolumeSnapshotBeta
VolumeSnapshot test - source namespace: snapshot-test-source-19461 - target namespace: snapshot-test-target-19461
PASS: snapshot_test.go:343: SnapshotTestSuite.TestVolumeSnapshotBeta    9.524s

START: snapshot_test.go:177: SnapshotTestSuite.TestVolumeSnapshotClassCloneFake
PASS: snapshot_test.go:177: SnapshotTestSuite.TestVolumeSnapshotClassCloneFake  0.003s

START: snapshot_test.go:240: SnapshotTestSuite.TestVolumeSnapshotCloneFake
PASS: snapshot_test.go:240: SnapshotTestSuite.TestVolumeSnapshotCloneFake       0.005s

START: snapshot_test.go:133: SnapshotTestSuite.TestVolumeSnapshotFake
PASS: snapshot_test.go:133: SnapshotTestSuite.TestVolumeSnapshotFake    0.002s

START: snapshot_test.go:354: SnapshotTestSuite.TestVolumeSnapshotStable
VolumeSnapshot test - source namespace: snapshot-test-source-19461 - target namespace: snapshot-test-target-19461
PASS: snapshot_test.go:354: SnapshotTestSuite.TestVolumeSnapshotStable  23.756s

START: snapshot_test.go:128: SnapshotTestSuite.TearDownSuite
Failed to list snapshots, Namespace: snapshot-test-source-19461, Error: the server could not find the requested resource
Skipping deleting the namespace, Namespace: snapshot-test-source-19461
Failed to list snapshots, Namespace: snapshot-test-target-19461, Error: the server could not find the requested resource
Skipping deleting the namespace, Namespace: snapshot-test-target-19461
PASS: snapshot_test.go:128: SnapshotTestSuite.TearDownSuite     4.130s

OK: 10 passed, 1 skipped
--- PASS: Test (49.22s)
PASS
ok      github.com/kanisterio/kanister/pkg/kube/snapshot        49.244s

```

### CODE REVIEW

**Review by pavannd1 — COMMENTED** [review]
LGTM +1

File: pkg/kube/snapshot/snapshot_alpha.go [hunk 1]
```diff
@@ -113,18 +114,61 @@ func (sna *SnapshotAlpha) Create(ctx context.Context, name, namespace, pvcName s
...
+		vsRet.Spec.Source.PersistentVolumeClaimName = &vs.Spec.Source.Name
+	}
+	if vs.Spec.VolumeSnapshotClassName != "" {
+		vsRet.Spec.VolumeSnapshotClassName = &vs.Spec.VolumeSnapshotClassName
+	}
+	if vs.Spec.SnapshotContentName != "" {
+		vsRet.Spec.Source.VolumeSnapshotContentName = &vs.Spec.SnapshotContentName
+	}
+
+	if vs.Status == (v1alpha1.VolumeSnapshotStatus{}) {
```
Comment (bathina2) [inline pkg/kube/snapshot/snapshot_alpha.go:146]: Do you need to check if its nil? Or will it always be loaded with an empty object?

**Review by bathina2 — APPROVED** [review]
Looks great!

File: pkg/kube/snapshot/snapshot_alpha.go (hunk 1 above)
Comment (PrasadG193) [reply pkg/kube/snapshot/snapshot_alpha.go:146]: We do not always get an empty object, if we do, we return from here

**Pull request merged**